assets/js/utils/attributes-query.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import { sortBy, map } from 'lodash';
+
+/**
+ * Given a query object, removes an attribute filter by a single slug.
+ * @param {object} query Current query object.
+ * @param {function} setQuery Callback to update the current query object.
+ * @param {object} attribute An attribute object.
+ * @param {string} slug Term slug to remove.
+ */
+export const removeAttributeFilterBySlug = (
+	query = [],
+	setQuery = () => {},
+	attribute,
+	slug = ''
+) => {
+	// Get current filter for provided attribute.
+	const currentQuery = query.filter(
+		( item ) => item.attribute === attribute.taxonomy
+	)[ 0 ];
+
+	if (
+		! currentQuery ||
+		! currentQuery.slug ||
+		! currentQuery.slug.includes( slug )
```

Comment:
Is `currentQuery.slug` an array or a string here (`includes` is a valid function on either's prototype) - I realize from later logic it appears it's assumed it's an array.  Reason I point this out is I think we need to tighten this conditional a bit because `currentQuery.slug` _could_ be a string - eg:

```js
const someString = 'mobile';
// true
someString.includes( 'mob' );
```

So maybe add a `! Array.isArray( currentQuery.slug )` check here?